### ISSUE TITLE
keyboard navigation not working in dialog on macOS

### ISSUE THREAD

**Issue by tessus** [electron/electron#21710]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v7.1.8
* **Operating System:**
  * macOS 10.14.6 (but I think it doesn't matter. I think it happens on all macOS versions)
* **Last Known Working Electron version:**
  * none that I know of

### Expected Behavior
When the dialog shows up, I should be able to switch to `Ok` with either the TAB key or the right arrow key. And I should be able to confirm with `Enter` or `Space`.

### Actual Behavior

The button `Cancel` is selected (thanks to fix #21642 ), but I can't select anything else nor can I dismiss the dialog by pressing `Enter` or `Space`.

<img width="552" alt="image" src="https://user-images.githubusercontent.com/223439/72021741-b5ef3200-323c-11ea-83a3-d82e2f0df65b.png">

### To Reproduce

electron-fiddle: https://gist.github.com/7b813289b0f9dab2aee48155415f0255

Minimal example:

`test.js`

```
const {app, BrowserWindow, dialog} = require('electron');

let mainWindow;

app.on('ready', () => {
	mainWindow = new BrowserWindow({
		height: 200,
		width: 440
	});

	mainWindow.loadURL('file://' + __dirname + '/index.html');

	const ok = dialog.showMessageBoxSync(mainWindow, {
		type: 'question',
		message: 'message here',
		buttons: ['Ok', 'Cancel'],
		defaultId: 1,
		cancelId: 1,
	});
});
```

`node_modules/electron/dist/Electron.app/Contents/MacOS/Electron test.js`

<img width="552" alt="image" src="https://user-images.githubusercontent.com/223439/72021741-b5ef3200-323c-11ea-83a3-d82e2f0df65b.png">

---

Context:

Same as #21633

**Comment by laurent22** [comment]
I get the feeling they are deprecating these synchronous dialog boxes, and in fact they made a breaking change to them in v7 (which as always they didn't document).

Perhaps we should switch to the async dialog methods as it seems to be what will be supported now.

**Comment by tessus** [comment]
The async dialog box seems to have the same problem though.

**Comment by tessus** [comment]
electron-fiddle: https://gist.github.com/7b813289b0f9dab2aee48155415f0255

**Comment by codebytere** [comment]
@tessus what do you expect to happen here in terms of enter and escape key equivalent logic? The default key is typically bound to `Enter`, but you've set the default and the cancel id to the same; cancellation key equivalent takes precedent which is why only `Escape` works here.

macOS makes it such that any given NSButton can have only 1 key equivalent, so it's impossible to have `Enter` and `Escape` trigger the same behavior.

>  I should be able to switch to Ok with either the TAB key or the right arrow key. 

This is not Electron's functional jurisdictional. You need to change the setting yourself in System Preferences. 

<img width="598" alt="Screen Shot 2020-01-13 at 2 08 41 PM" src="https://user-images.githubusercontent.com/2036040/72296599-dfc6a100-360e-11ea-96de-c35a63084414.png">

Testing your fiddle locally, when I change that setting, focus the non-default button, and press the spacebar, I see the expected behavior you referenced.

**Comment by tessus** [comment]
@codebytere thanks for looking into this.

> macOS makes it such that any given NSButton can have only 1 key equivalent, so it's impossible to have Enter and Escape trigger the same behavior.

Hmm, I don't think this is entirely correct: e.g. Open Finder and set `Show warning before changing extension`. Then change an extension. A dialog shows up and the default is `keep file.ext`. I can select keep by using `Enter`, `Space`, and I can cancel the dialog by using `Esc`.
However when I use `Tab` to go to the other button, I have to use `Space` as `Enter` does not work anymore. (`Enter` only works on the default button.)
And, btw, the buttons show up in the Touchbar, but I guess that's a different story. Although it would be awesome if Electron were to to do the same. ;-)

In Electron `Enter` does not work for the default button, but I guess I can live with that. The more important question is why that is and why it works in Finder. If people are used to use `Enter`, they will be puzzled why it doesn't work in Electron.

> This is not Electron's functional jurisdiction. You need to change the setting yourself in System Preferences.

Yes, my bad. You are absolutely right.

So basically I think we can close the issue. Thanks again for taking time to look into it.

**Comment by codebytere** [comment]
Thanks for the follow-up, and glad it seems resolved - I'll see what i can do about touch-bar stuff in the future :)
